xmppserver/src/main/java/org/jivesoftware/openfire/muc/spi/MultiUserChatServiceImpl.java

```diff
@@ -594,24 +594,33 @@ private void checkForTimedOutUsers() {
         @Override
         public void run() {
             try {
-                logConversation();
+                logConversationBatch();
             }
             catch (final Throwable e) {
                 Log.error(LocaleUtils.getLocalizedString("admin.error"), e);
             }
         }
     }
 
-    private void logConversation() {
+    private void logConversationBatch() {
         ConversationLogEntry entry;
         boolean success;
+
+        final LinkedList<ConversationLogEntry> batchList = new LinkedList<>();
+
```

Comment:
The next line limits the number of statements in a batch to log_batch_size

Reply:
It has been tested in a clustered environment with a batch size of 300 per node, and there are no message IDs duplication. log_batch_size can be set to a value greater than SequenceManager#blockSize without problems (see previous comment)

Reply:
Ah, apologieds, mis-understood what was going on.